x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsPendingDeletionsIntegTests.java

```diff
@@ -0,0 +1,575 @@
+                    Strings.EMPTY_ARRAY,
+                    true,
+                    randomFrom(MountSearchableSnapshotRequest.Storage.values())
+                ),
+                new ActionListener<>() {
+                    @Override
+                    public void onResponse(RestoreSnapshotResponse response) {
+                        assertThat(response.getRestoreInfo().successfulShards(), greaterThan(0));
+                        assertThat(response.getRestoreInfo().failedShards(), equalTo(0));
+                        mounts.put(mount, snapshotInfo.snapshotId());
+                        mountLatch.countDown();
+                    }
+
+                    @Override
+                    public void onFailure(Exception e) {
+                        final AssertionError error = new AssertionError("error during mount", e);
+                        logger.error("test failed", error);
+                        mountLatch.countDown();
+                        throw error;
+                    }
+                }
+            );
+        }
+        mountLatch.await();
+
+        if (randomBoolean()) {
+            // we can't delete a repository that is used by searchable snapshot indices but we can update it
+            assertAcked(
+                clusterAdmin().preparePutRepository(repository)
+                    .setType(FsRepository.TYPE)
+                    .setVerify(randomBoolean())
+                    .setSettings(randomBoolean() ? repositorySettings : repositorySettings.put("dummy", randomInt()))
+            );
+        }
+
+        final CyclicBarrier startThreads = new CyclicBarrier(2);
+        final Thread deleteAllSnapshotsThread = new Thread(() -> {
+            try {
+                startThreads.await();
+                final CountDownLatch latch = new CountDownLatch(snapshots.size());
+                for (SnapshotInfo snapshot : snapshots) {
+                    client().admin()
+                        .cluster()
+                        .prepareDeleteSnapshot(repository, snapshot.snapshotId().getName())
+                        .execute(ActionListener.wrap(latch::countDown));
+
+                }
+                latch.await();
+            } catch (Exception e) {
+                throw new AssertionError(e);
+            }
+        });
+        deleteAllSnapshotsThread.start();
+
+        final Thread deleteAllMountedIndicesThread = new Thread(() -> {
+            try {
+                ensureGreen("mount-*");
+                startThreads.await();
+                final Set<String> mountedIndices = new HashSet<>(mounts.keySet());
+                do {
+                    List<String> deletions = randomSubsetOf(Math.max(1, mountedIndices.size() - 1), mountedIndices);
+                    assertAcked(client().admin().indices().prepareDelete(deletions.toArray(String[]::new)));
+                    deletions.forEach(mountedIndices::remove);
```

Comment:
Is this not the same as `mountedIndices.removeAll(deletions)`?

Reply:
AbstractSet#removeAll's javadoc indicates that the method behaves differently depending of the sizes of the sets that are involved, but in this case `deletions` should be smaller that `mountedIndices` so `removeAll` should operate similarly to `deletions.forEach()`.